temporalcli/commandsgen/commands.yml

```diff
@@ -647,6 +682,119 @@ commands:
         description: Reason for terminating the batch job.
         required: true
 
+  - name: temporal config
+    summary: Manage config files (EXPERIMENTAL)
+    description: |
+      Config files are TOML files that contain profiles, with each profile
+      containing configuration for connecting to Temporal. 
+
+      ```
+      temporal config set \
+          --prop address \
+          --value us-west-2.aws.api.temporal.io:7233
+      ```
+
+      The default config file is at `$CONFIG_PATH/temporal/temporal.toml` where
+      `$CONFIG_PATH` is defined as `$HOME/.config` on Unix,
+      `$HOME/Library/Application Support` on macOS, and `%AppData%` on Windows.
+      This can be overridden with the `TEMPORAL_CONFIG_FILE` environment
+      variable or `--config-file`.
+
+      The default profile is `default`. This can be overridden with the
+      `TEMPORAL_PROFILE` environment variable or `--profile`.
+    docs:
+      description-header: >-
+        Temporal CLI 'config' commands allow the getting, setting, deleting, and
+        listing of configuration properties for connecting to Temporal.
+      keywords:
+        - cli reference
+        - command-line-interface-cli
+        - configuration
+        - config
+        - config delete
+        - config get
+        - config list
+        - config set
+        - environment
+        - temporal cli
+
+  - name: temporal config delete
+    summary: |
+      Delete a config file property or an entire config profile (EXPERIMENTAL)
+    description: |
+      Remove a full profile entirely _or_ remove a property within a profile.
+      When deleting an entire profile, the `--profile` must be set explicitly.
+
+      ```
+      temporal env delete \
+          --profile my-profile
+      ```
```

Comment:
IMO adding an explicit `--delete-whole-profile` or something might be good in that instance, but, not hugely concerned.

Reply:
:+1: Yeah this just matches what we do for `env` today, but I do agree. Actually this should be a top-level command called `delete-profile` so `delete` can always require the `--prop`.

Reply:
Moved to its own command. Leaving conversation open for others that may want to see that this changed.